pmm-app/src/pmm-settings/components/SignUp/SignUp.tsx

```diff
@@ -0,0 +1,126 @@
+import React, {
+  FC,
+  useState
+} from 'react';
+import { Form as FinalForm, Field as FinalField, FormRenderProps } from 'react-final-form';
+import {
+  Button,
+  LinkButton,
+  useTheme,
+} from '@grafana/ui';
+import { showSuccessNotification, showErrorNotification } from 'shared/components/helpers';
+import { ButtonWithSpinner } from 'shared/components/Form';
+import { getStyles } from './SignUp.styles';
+import { Messages } from './SignUp.messages';
+import { InputFieldAdapter, CheckboxFieldAdapter } from './FieldAdapters/FieldAdapters';
+import { TERMS_OF_SERVICE_URL, PRIVACY_POLICY_URL, NOTIFICATION_SETTINGS_URL } from './SignUp.constants';
+import { required, validEmail, composeValidators } from './SignUp.validators';
```

Comment:
I guess that this functionality should be moved to shared, we already have something for validation there

Reply:
good idea!

Reply:
Done